src/main/java/com/redhat/devtools/intellij/tektoncd/ui/editors/PipelineRunGraphUpdater.java

```diff
@@ -0,0 +1,84 @@
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.tektoncd.ui.editors;
+
+import com.mxgraph.view.mxGraph;
+import com.redhat.devtools.intellij.common.utils.DateHelper;
+import io.fabric8.tekton.pipeline.v1beta1.PipelineRun;
+import io.fabric8.tekton.pipeline.v1beta1.PipelineRunTaskRunStatus;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Optional;
+
+public class PipelineRunGraphUpdater extends AbstractPipelineGraphUpdater<PipelineRun> {
+
+    @Override
+    public PipelineRun adapt(String content) {
+        try {
+            return MAPPER.readValue(content, PipelineRun.class);
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public void update(PipelineRun content, mxGraph graph) {
+        if (content.getStatus() != null) {
+            update(content, content.getStatus().getPipelineSpec(), graph);
+        }
+    }
+
+    @Override
+    protected String getNodeStyle(PipelineRun content, Node node) {
+        String style = super.getNodeStyle(content, node);
+        if (node.type == Type.TASK) {
+            boolean succeeded = isTaskSucceeded(content, node.name);
+            if (succeeded) {
+                style += "fillColor=green;";
+            } else {
+                style += "fillColor=red;";
+            }
+        }
+        return style;
+    }
+
+    @Override
+    protected String getNodeLabel(PipelineRun content, Node node) {
+        String label = super.getNodeLabel(content, node);
+        if (node.type == Type.TASK) {
+            Optional<PipelineRunTaskRunStatus> taskStatus = getTaskStatus(content, node.name);
+            if (taskStatus.isPresent() && taskStatus.get().getStatus() != null &&
+                    StringUtils.isNotBlank(taskStatus.get().getStatus().getStartTime()) &&
+                    StringUtils.isNotBlank(taskStatus.get().getStatus().getCompletionTime())) {
+                label += "\n" + DateHelper.humanizeDate(Instant.parse(taskStatus.get().getStatus().getStartTime()),
+                        Instant.parse(taskStatus.get().getStatus().getCompletionTime()));
+            }
+        }
+        return label;
+    }
+
+    private boolean isTaskSucceeded(PipelineRun content, String name) {
+        boolean succeeded = false;
+        Optional<PipelineRunTaskRunStatus> taskStatus = getTaskStatus(content, name);
+        if (taskStatus.isPresent()) {
+            succeeded = taskStatus.get().getStatus().getConditions().stream().filter(condition -> "True".equals(condition.getStatus())).findFirst().isPresent();
```

Comment:
Here the getStatus().getConditions() can be null and make the rest fail. An example is if you use this https://github.com/tektoncd/pipeline/blob/master/examples/v1beta1/pipelineruns/conditional-pipelinerun.yaml and you click `Open In Editor` on the pipelineRun when it's still executing.

The error i face is:
```
ava.lang.NullPointerException
	at com.redhat.devtools.intellij.tektoncd.ui.editors.PipelineRunGraphUpdater.isTaskSucceeded(PipelineRunGraphUpdater.java:75)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.PipelineRunGraphUpdater.getNodeStyle(PipelineRunGraphUpdater.java:46)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.PipelineRunGraphUpdater.getNodeStyle(PipelineRunGraphUpdater.java:24)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.AbstractPipelineGraphUpdater.createVertex(AbstractPipelineGraphUpdater.java:112)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.AbstractPipelineGraphUpdater.generateGraph(AbstractPipelineGraphUpdater.java:84)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.AbstractPipelineGraphUpdater.generateGraph(AbstractPipelineGraphUpdater.java:85)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.AbstractPipelineGraphUpdater.generateGraph(AbstractPipelineGraphUpdater.java:85)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.AbstractPipelineGraphUpdater.generateGraph(AbstractPipelineGraphUpdater.java:79)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.AbstractPipelineGraphUpdater.update(AbstractPipelineGraphUpdater.java:73)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.PipelineRunGraphUpdater.update(PipelineRunGraphUpdater.java:38)
	at com.redhat.devtools.intellij.tektoncd.ui.editors.PipelineRunGraphUpdater.update(PipelineRunGraphUpdater.java:24)
```